tictactoe_engine.py

```diff
@@ -0,0 +1,91 @@
+from enum import Enum
+
+state_list=[]
```

Comment:
There should not be a global variable. The states can be a member of the game engine or interface.

Reply:
yes this can be done, I did that first but wasn't working fine
Now it's working